l10n_it_ddt/models/sale.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2014 Abstract (http://www.abstract.it)
+# Copyright Davide Corio <davide.corio@abstract.it>
+# Copyright 2014-2018 Agile Business Group (http://www.agilebg.com)
+# Copyright 2015 Apulia Software s.r.l. (http://www.apuliasoftware.it)
+# Copyright Francesco Apruzzese <f.apruzzese@apuliasoftware.it>
+# Copyright 2018 Simone Rubino - Agile Business Group
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models, api
+
+
+class SaleOrder(models.Model):
+
+    _inherit = 'sale.order'
+
+    @api.multi
+    def _compute_ddt_ids(self):
+        for so in self:
+            ddt_ids = []
+            for picking in so.picking_ids:
+                for ddt in picking.ddt_ids:
+                    ddt_ids.append(ddt.id)
+            so.ddt_ids = ddt_ids
+
+    carriage_condition_id = fields.Many2one(
+        'stock.picking.carriage_condition', string='Carriage Condition')
+    goods_description_id = fields.Many2one(
+        'stock.picking.goods_description',
+        string='Description of Goods')
+    transportation_reason_id = fields.Many2one(
+        'stock.picking.transportation_reason',
+        string='Reason for Transportation')
+    transportation_method_id = fields.Many2one(
+        'stock.picking.transportation_method',
+        string='Method of Transportation')
+    ddt_carrier_id = fields.Many2one(
+        'res.partner', string='Carrier')
+    parcels = fields.Integer('Parcels')
+    weight = fields.Float(string="Weight")
+    gross_weight = fields.Float(string="Gross Weight")
+    volume = fields.Float('Volume')
+    ddt_ids = fields.Many2many(
+        'stock.picking.package.preparation',
+        string='Related TDs',
+        compute='_compute_ddt_ids')
+    create_ddt = fields.Boolean('Automatically create the TD')
+    ddt_invoicing_group = fields.Selection(
+        [('nothing', 'One TD - One Invoice'),
+         ('billing_partner', 'Billing Partner'),
+         ('shipping_partner', 'Shipping Partners'),
```

Comment:
Questa in teoria dovrebbe essere al singolare come in https://github.com/OCA/l10n-italy/pull/836/files#diff-841730856eedc87ed5e34d0265652bbeR30

Shipping Partner

Reply:
Corretto